src/pages/Auth/index.tsx

```diff
@@ -1,40 +1,66 @@
 import { ReactComponent as GoogleIcon } from 'assets/svg/google.svg';
 import { ReactComponent as NaverIcon } from 'assets/svg/naver.svg';
 import { ReactComponent as KakaoIcon } from 'assets/svg/kakao.svg';
-import { useForm, SubmitHandler } from 'react-hook-form';
-import { Link } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { Link, useNavigate } from 'react-router-dom';
 import AuthTitle from 'components/Auth/AuthTitle';
 import Copyright from 'components/Auth/Copyright';
 import cn from 'utils/ts/classNames';
+import { login } from 'api/user';
+import sha256 from 'sha256';
+import makeToast from 'utils/ts/makeToast';
 import styles from './LoginPage.module.scss';
 
-interface IFormInput {
+interface LoginFormInput {
   id: string;
   pw: string;
   checkbox: boolean;
 }
 
+const useLoginRequest = () => {
+  const navigate = useNavigate();
+  const submitLogin = async ({ id, pw, checkbox }: LoginFormInput) => {
+    try {
+      const { data } = await login({
+        account: id,
+        password: sha256(pw),
+      });
+
+      if (data) {
+        sessionStorage.setItem('accessToken', data.accessToken);
+        if (checkbox) localStorage.setItem('refreshToken', data.refreshToken);
+        navigate('/');
+      }
+    } catch (e) {
+      // TODO - 401 에러를 제외한 나머지 핸들링 (500 서버에러 등)
+      makeToast('error', '올바르지 않은 계정 정보입니다!');
+    }
+  };
+
+  return submitLogin;
+};
+
 function LoginPage(): JSX.Element {
```

Comment:
급궁금... 요거 반환타입 붙여주는 이유가 어떤건가요?

```suggestion
function LoginPage() {
```

Reply:
아마... create react-app 을 했을 때 기본 템플릿의 `App.tsx`가 저렇게 정의되어있어서 초기에 따라간 것 같습니다
알아서 추론되니 필요없는 구문이긴 할테지만요..ㅋㅋㅋ

Reply:
아하 글쿤요 ㅎㅎ 특별한 이유가 있는줄 알았어요~
사실 요렇게 return type 을 명시하면 ts compiler 의 성능이 올라간다고 해요~ [링크](https://github.com/microsoft/TypeScript/wiki/Performance#using-type-annotations)